lib/chef/cookbook_version.rb

```diff
@@ -532,7 +535,7 @@ def self.chef_server_rest
     end
 
     def chef_server_rest
-      self.class.chef_server_rest
+      @chef_server_rest ||= Chef::REST.new(Chef::Config[:chef_server_url])
```

Comment:
Except those class methods with global state are terrible, I'd rather go the other direction and make all the class methods call new to create an instance to do their work.